DigitalLearningSolutions.Data/Services/CourseService.cs

```diff
@@ -405,6 +413,28 @@ Customisation customisation
             return filteredApplications.OrderBy(a => a.ApplicationName);
         }
 
+        public IEnumerable<ApplicationWithSections> GetApplicationsByBrandId(int brandId)
+        {
+            var numRecordsByApplicationId =
+                courseDataService.GetNumsOfRecentProgressRecordsForBrand(brandId, clockService.UtcNow.AddMonths(-3));
+            var applications = courseDataService.GetApplicationsByBrandId(brandId);
+            double maxPopularity = numRecordsByApplicationId.Any() ? numRecordsByApplicationId.First().Value : 0;
+            var applicationsWithSections = applications.Select(
+                application => new ApplicationWithSections(
+                    application,
+                    sectionService.GetSectionsThatHaveTutorialsAndPopulateTutorialsForApplication(
+                        application.ApplicationId
+                    ),
+                    maxPopularity == 0
+                        ? 0
+                        : (numRecordsByApplicationId.ContainsKey(application.ApplicationId)
+                            ? numRecordsByApplicationId[application.ApplicationId] / maxPopularity
+                            : 0)

```

Comment:
Could we make this a bit clearer by doing this in separate stages? Nested ternary statements are a bit hard to follow.

Reply:
I've realised I am able to do away with the second statement by using .GetValueOrDefault()